Symfony/CS/Tests/FileCacheManagerIntegrationTest.php

```diff
@@ -0,0 +1,129 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Symfony\CS\Tests;
+
+use Symfony\Component\Filesystem\Filesystem;
+use Symfony\Component\Finder\SplFileInfo as FinderSplFileInfo;
+use Symfony\CS\FileCacheManager;
+use Symfony\CS\Fixer;
+
+class FileCacheManagerIntegrationTest extends \PHPUnit_Framework_TestCase
+{
+    protected function setUp()
+    {
+        $this->fs = new Filesystem();
+        $this->rootDir = __DIR__.DIRECTORY_SEPARATOR.'Fixtures'.DIRECTORY_SEPARATOR.'FileCacheManagerIntegrationTest';
+
+        $this->fs->mkdir($this->rootDir);
+        $this->isCacheAvailable = $this->isCacheAvailable();
+
+        // Never ever rely on last test results
+        $this->clearFileCacheManagerIntegrationTestFiles();
+
+        $this->customConfigDir = $this->rootDir.DIRECTORY_SEPARATOR.'custom_config_dir';
+
+        $this->files = array(
+            'one' => array(
+                'filename' => $this->rootDir.DIRECTORY_SEPARATOR.'path_one'.DIRECTORY_SEPARATOR.'CommonFilename.php',
+                'content' => "<?\n// one\n",
+            ),
+            'two' => array(
+                'filename' => $this->rootDir.DIRECTORY_SEPARATOR.'path_two'.DIRECTORY_SEPARATOR.'CommonFilename.php',
+                'content' => "<?\n// two\n",
+            ),
+        );
+
+        $this->fs->mkdir($this->rootDir);
+        $this->fs->mkdir($this->customConfigDir);
+
+        array_map(function (array $file) {
```

Comment:
I believe it's a matter of taste, but personally, I prefer to use `array_map()` only when I am interested in the return value of it, otherwise I would use `array_walk()`, for example.